src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncHelpers.CoreCLR.cs

```diff
@@ -140,7 +140,9 @@ public static partial class AsyncHelpers
         private struct RuntimeAsyncAwaitState
         {
             public Continuation? SentinelContinuation;
+            public ICriticalNotifyCompletion? CriticalNotifier;
             public INotifyCompletion? Notifier;
```

Comment:
If someone calls `AsyncHelpers.AwaitAwaiter`, then that should result in call to `INotifyCompletion.OnCompleted`, but today if the awaiter also implements `ICriticalNotifyCompletion` it will instead end up calling `ICriticalNotifyCompletion.UnsafeOnCompleted`. This is a fix for that.